cluster-autoscaler/cloudprovider/oci/nodepools/oci_manager.go

```diff
@@ -542,12 +545,22 @@ func (m *ociManagerImpl) buildNodeFromTemplate(nodePool *oke.NodePool) (*apiv1.N
 		})
 	}
 
+	freeformTags, err := m.ociTagsGetter.GetFreeformTags(nodePool)
```

Comment:
Given the discussion in the [other thread](https://github.com/kubernetes/autoscaler/pull/6607#discussion_r1534710107), I would expect [our shape struct](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/oci/common/oci_shape.go#L53) to have a new field, say, `EphemeralStorageInGB`, which may or may not be set after the call to `m.ociShapeGetter.GetNodePoolShape(nodePool)` based on [disk info](https://docs.oracle.com/en-us/iaas/Content/Compute/References/computeshapes.htm) of the node shape.

It is at this point that `buildNodeFromTemplate()` would set the value of `node.Status.Capacity[apiv1.ResourceEphemeralStorage]` to either `shape.EphemeralStorageInGB` or the value from the free-form tag.

Does that make sense?

Reply:
We get the LocalDiskSize of a shape from the Core Service/Compute API - ([LINK](https://github.com/oracle/oci-go-sdk/blob/master/core/shape.go#L70)).
But this is not part of the [NodeShapeConfig](https://github.com/oracle/oci-go-sdk/blob/master/containerengine/node_shape_config.go).
We would need to work on adding the changes to oci-go-sdk so that the NodeShapeConfig has the LocalDiskSize, but this would be a whole api level change and would involve a lot of time and effort.

Reply:
OK. Since there are _two_ implementations of this provider (and because the core API may evolve to offer ephemeral storage), I think it still makes sense to expose this attribute to both implementation using our [common.Shape struct](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/oci/common/oci_shape.go#L53) alongside the other capacity attributes (CPU, memory, GPU). 

So, for now, you still fetch the value from the free-form tag using a  `common.getEphemeralResourceRequestsInBytes(freeformTags)` function, but instead of setting a local / private variable, you'd set [shape.EphemeralStorageGB](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/oci/common/oci_shape.go#L53).


cluster-autoscaler/cloudprovider/oci/common/oci_shape.go

```
type Shape struct {
	Name             string
	CPU              float32
	GPU              int
	MemoryInBytes    float32
++	EphemeralStorageGB float32
}
```


As a followup MR, we'd need to ensure the _instancepool_ implementation is also using the `common.getEphemeralResourceRequestsInBytes(freeformTags)` function to check for ans set `shape.EphemeralStorageGB`

Reply:
Agreed. I have updated the PR with the above suggested changes.
Please take a look.